scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# source-maintenance plugin to standardize the header guards
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+
+if ($filename !~ /\.h$/i || $filename =~ /^scripts\/boilerplate.h/i) {
+    while (<>) { print; }
+    exit(0);
+}
+
+my @accumulate=();
+my $current_guard = undef;
+
+my $first_ifdef_pos = undef;
```

Comment:
```suggestion
my $first_ifndef_pos = undef;
```

or, given the current (complex) code structure:

```suggestion
my $guard_ifdef_pos = undef;
```

... or something like that. I do not insist on this change, but I find the current "ifdef/ifndef" mismatch rather confusing.

Reply:
Good point. Replacing with $first_ifndef_pos